### PR TITLE
Improve config.m4

### LINKED ISSUE #17: Better way to solve _ZNSt9exceptionD2Ev: symbol not found
See https://github.com/crazyxman/simdjson_php/commit/08ac37325da92c02415384ec83a72ba302cbc8ae for more details.

@remicollet Any idea how to fix it?

### PR DESCRIPTION
**Pull request by remicollet**
Fixes #17 

Second commit is only a cleanup / simplify

Pick the one(s) you want.

### CODE REVIEW

File: config.m4 [hunk 1]
```diff
@@ -10,7 +10,6 @@ PHP_ARG_ENABLE(simdjson, whether to enable simdjson, [ --enable-simdjson   Enabl
 
 if test "$PHP_SIMDJSON" != "no"; then
 
-  CXXFLAGS=" -march=native -O2 -std=c++17 "
```
Comment (remicollet) [inline config.m4:13]: Drop this to honor system build flag
Can de add in PHP_NEW_EXTENSION as "additional" flags

File: config.m4 [hunk 2]
```diff
@@ -41,15 +40,12 @@ if test "$PHP_SIMDJSON" != "no"; then
...
-  PHP_SUBST(SIMDJSON_SHARED_LIBADD)
-  PHP_ADD_LIBRARY(stdc++, 1, SIMDJSON_SHARED_LIBADD)
-
   AC_DEFINE(HAVE_SIMDJSON, 1, [whether simdjson is enabled])
   PHP_NEW_EXTENSION(simdjson,
       simdjson.cpp                        \
       src/bindings.cpp                    \
       src/simdjson.cpp,
-    $ext_shared)
+    $ext_shared,, -std=c++17, cxx)
```
Comment (remicollet) [inline config.m4:37]: Invoke proper linker command, so add needed libraries (g++ always add libstdc++)

**Comment by crazyxman** [comment]
@remicollet  That's great. Thank you!

**Comment by sandrokeil** [comment]
@remicollet Good work. :+1: Can you please rebase your PR with the latest changes of master, please? I've fixed the pipeline.

**Comment by remicollet** [comment]
Rebased

**Comment by remicollet** [comment]
And CI seems happy :)

**Pull request merged**